sentry.client.config.ts

```diff
@@ -12,17 +12,22 @@ Sentry.init({
   dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,
 
   // Adjust this value in production, or use tracesSampler for greater control
-  tracesSampleRate: 1,
+  tracesSampleRate: ["development", "heroku"].includes(process.env.NODE_ENV)
```

Comment:
Q: Should we extract this to some shared `isDev()`-esque utility function since we're using the pattern at least 5&times;:

```sh
git grep -nP "\bheroku\b" -- ':!*.md' ':!.circleci/*' ':!.env*' ':!src/db/migrations/*' | cat

next.config.js:124:    const noindexEnvs = ["dev", "development", "heroku", "stage"];
sentry.client.config.ts:15:  tracesSampleRate: ["development", "heroku"].includes(process.env.NODE_ENV)
sentry.client.config.ts:20:  debug: ["development", "heroku"].includes(process.env.NODE_ENV),
sentry.client.config.ts:26:  replaysSessionSampleRate: ["development", "heroku"].includes(
sentry.server.config.ts:15:  tracesSampleRate: ["development", "heroku"].includes(process.env.NODE_ENV)
sentry.server.config.ts:20:  debug: ["development", "heroku"].includes(process.env.NODE_ENV),
src/appConstants.js:68:// @ts-ignore TS thinks NODE_ENV can't be "heroku". Shut up TS, you're drunk!
src/appConstants.js:69:if (!process.env.SERVER_URL && (process.env.NODE_ENV) === 'heroku') {
src/controllers/dockerflow.js:26:   if (AppConstants.NODE_ENV === 'heroku') {
src/db/knexfile.js:10:if (NODE_ENV === 'heroku') {
src/middleware/noSearchEngineIndex.js:8:const noindexEnvs = ['dev', 'heroku', 'stage']
```

Reply:
Yes I think it's worth considering, but I would note that in the above examples:

- we sometimes want to include stage, but not always
- we're moving from `dev` to the more standard `development`
- most (but not all) of the code in `./src/` will be replaced by the NextJS app in `./src/app`

NextJS suggests this for environment names:
https://nextjs.org/docs/messages/non-standard-node-env

> The greater React ecosystem treats NODE_ENV as a convention, only permitting three (3) values:
>
>   production: When your application is built with next build
>   development: When your application is run with next dev
>   test: When your application is being tested (e.g. jest)

Regarding staging:

> If you need the concept of different environments in your application, e.g. staging, you should use a different environment variable name like APP_ENV.

So I would prefer to clean this up a bit and come back to it, I'll file an issue in our backlog.

Reply:
After trying this for a bit, this makes the logs way too noisy even in dev. I'm going to just set it to `false` by default, thanks for bringing this up though! Will follow up in the other tickets.